client/service/src/lib.rs

```diff
@@ -451,22 +462,32 @@ fn build_network_future<
 			(status, state)
 		});
 
+		// Processing DHT events.
+		while let Ok(Async::Ready(Some(event))) = event_stream.poll() {
```

Comment:
Given that the authority discovery module has a reference to the network service, it could register its own event stream, thus the `build_network_future` function does not need to be aware of the authority discovery module no more, correct? Happy to do that in a follow up.

Reply:
Exactly. I was leaving that for a later refactoring, but that's totally feasible.

Reply:
Would be fixed with https://github.com/paritytech/substrate/pull/4344.